### PR TITLE
Balances chart

### PR DESCRIPTION
**Pull request by Code47X**
In this PR:

- Finished up Balances graph
- Changed up styling and color generation for better visibility with unknown amount of records
- Slight cleanup to components - moved date formatting back into the graph models.
- When hovering a specific account in the legend, the others will fade out slightly.
- Tooltips will now auto sort order (highest Y-axis value to lowest)

Man that intro animation on the balance graph is pretty satisfying.

https://user-images.githubusercontent.com/5169367/112665357-f9759200-8e28-11eb-8613-b6ffe3782919.mp4

### CODE REVIEW

File: app/javascript/components/BalanceGraph.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+const TooltipContent = ({ active, payload }) => {
+  if (active && payload && payload.length) {
+    const data = payload[0].payload;
+
+    return (
+      <div
+        className="custom-tooltip"
+        style={{ backgroundColor: 'rgba(0, 0, 0, 0.5)', padding: 10, textAlign: 'center', borderRadius: 8 }}
```
Comment (Chazz-Michaels) [inline app/javascript/components/BalanceGraph.jsx:19]: The tooltip text can be kind of hard to read here. I'm guessing this was a preexisting convention but I played around with the tooltip background color a bit and I'm wondering that you think of this:

Original tooltip:
tooltip_original_bg

Proposed tooltip lighter background (`rgba(210, 210, 210, 0.5)`) with date as black text color (`#000000`):
tooltip_proposed_bg_210_210_210_0 5

**Review by Chazz-Michaels — COMMENTED** [review]
I think if you try to visit the `/balances` page without a balances CSV loaded first, the page breaks

File: app/javascript/components/BalanceGraph.jsx (hunk 1 above)
Comment (Chazz-Michaels) [reply app/javascript/components/BalanceGraph.jsx:19]: There's a cool, new-ish CSS property called `backdrop-filter` that we may be able to leverage in the future for adding background blur to these tooltips without needing a parent element to house the blur property, and that could help the text really pop here. But at the moment, it's lacking major browser support.
https://developer.mozilla.org/en-US/docs/Web/CSS/backdrop-filter

Comment (Code47X) [reply app/javascript/components/BalanceGraph.jsx:19]: Agreed this looks better! I'll throw in the change.

**Pull request merged**